subsys/nrf_security/src/drivers/cracen/cracenpsa/src/cracen_mac_cmac.c

```
@@ -76,14 +76,15 @@ psa_status_t cracen_cmac_update(cracen_mac_operation_t *operation, const uint8_t
 	/* The state can only be resumed if is not the first time data are
 	 * processed.
 	 */
-	if (!operation->is_first_block) {
+	if (cracen_cmac_supports_multi_part() && !operation->is_first_block) {
```

Are you certain that this will fail for multipart? Its a bit tough to understand what is happening here without failing early. Have you considered to create a static function which is only called when cracen_cmac_supports_multi_part is enabled? It simplifies the logic and the return code behavior in my opinion

---

Yes, if someone tries to call the multi-part functions they will return `PSA_ERROR_NOT_SUPPORTED` early in `mac.c`. Here we only need to make sure to not use context switching if it's not supported.

---

Discussed privately and agreed on changing the approach.